### PR TITLE
Cmake updates attempt 2

### PR DESCRIPTION
**Pull request by mathstuf**
Reapplication of #6368.

### CODE REVIEW

**Comment by mathstuf** [comment]
See https://github.com/compiler-explorer/infra/pull/1295 for infra updates.

**Comment by mathstuf** [comment]
I see no message in CI about what the failure might be :/ .

**Comment by RubenRBS** [comment]
You have a danging compiler id :)

https://github.com/compiler-explorer/compiler-explorer/actions/runs/8991939125/job/24700607237?pr=6452#step:5:598

[Here](https://github.com/compiler-explorer/compiler-explorer/pull/6452/files#diff-217ce3936f1e1cadf1d0b1b0b7096031c69951100ff5513ea440090507e35b74R9) it's `cmake-3_26_4`,
but in the compiler list [here](https://github.com/compiler-explorer/compiler-explorer/pull/6452/files#diff-217ce3936f1e1cadf1d0b1b0b7096031c69951100ff5513ea440090507e35b74R1), you've only set it as `cmake-3_26`

**Comment by mathstuf** [comment]
Thanks; fixed.

**Review by mattgodbolt — COMMENTED** [review]
LGTM! Just waiting on minor changes in the installer but this seems to be ok. Thanks so much for your patience.

File: etc/config/cmakescript.amazon.properties [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
 defaultCompiler=cmake-3_28_0
 
 compiler.cmake-3_26_1.exe=/opt/compiler-explorer/cmake-v3.26.1/bin/cmake
 compiler.cmake-3_26_1.semver=3.26.1
 
+compiler.cmake-3_26_4.exe=/opt/compiler-explorer/cmake-v3.26.4/bin/cmake
+compiler.cmake-3_26_4.semver=3.26.4
+
+compiler.cmake-3_26.exe=/opt/compiler-explorer/cmake-v3.26/bin/cmake
+compiler.cmake-3_26.semver=3.26.4
```
Comment (mattgodbolt) [inline etc/config/cmakescript.amazon.properties:13]: this is the same semver as above. It should be unique

File: etc/config/cmakescript.amazon.properties [hunk 2]
```diff
@@ -1,13 +1,37 @@
...
+
 compiler.cmake-3_27_9.exe=/opt/compiler-explorer/cmake-v3.27.9/bin/cmake
 compiler.cmake-3_27_9.semver=3.27.9
 
+compiler.cmake-3_27.exe=/opt/compiler-explorer/cmake-v3.27/bin/cmake
+compiler.cmake-3_27.semver=3.27.9
```
Comment (mattgodbolt) [inline etc/config/cmakescript.amazon.properties:19]: ditto

File: etc/config/cmakescript.amazon.properties [hunk 3]
```diff
@@ -1,13 +1,37 @@
...
+compiler.cmake-3_28.exe=/opt/compiler-explorer/cmake-v3.28/bin/cmake
+compiler.cmake-3_28.semver=3.28.4
+
+compiler.cmake-3_29_2.exe=/opt/compiler-explorer/cmake-v3.29.2/bin/cmake
+compiler.cmake-3_29_2.semver=3.29.2
+
+compiler.cmake-3_29.exe=/opt/compiler-explorer/cmake-v3.29/bin/cmake
+compiler.cmake-3_29.semver=3.29.2
+
+compiler.cmake-importstd.exe=/opt/compiler-explorer/cmake-importstd/bin/cmake
```
Comment (mattgodbolt) [inline etc/config/cmakescript.amazon.properties:36]:
```suggestion
compiler.cmake-importstd.exe=/opt/compiler-explorer/cmake-v3.29.20240506-g1ea8fa8/bin/ccmake
```

Maybe?

**Review by mattgodbolt — COMMENTED** [review]
Why the duplicated semvers?

I don't see a `cmake-importstd` executable after installing. I see:

```
May/18 22:25 admin-node~/infra (main|✔) 10.0s $ ls -dl /opt/compiler-explorer/cmake*
lrwxrwxrwx 1 root root   13 Dec 10 19:54 /opt/compiler-explorer/cmake -> cmake-v3.28.0
drwxr-xr-x 6 root root 6144 May 24  2022 /opt/compiler-explorer/cmake-v3.18.2
drwxr-xr-x 6 root root 6144 May 24  2022 /opt/compiler-explorer/cmake-v3.23.1
drwxr-xr-x 6 root root 6144 Nov 20  2022 /opt/compiler-explorer/cmake-v3.24.3
drwxr-xr-x 6 root root 6144 Jan 11  2023 /opt/compiler-explorer/cmake-v3.25.1
drwxr-xr-x 6 root root 6144 Mar 28  2023 /opt/compiler-explorer/cmake-v3.26.1
drwxr-xr-x 6 root root 6144 Apr 30 04:16 /opt/compiler-explorer/cmake-v3.26.4
drwxr-xr-x 6 root root 6144 Dec  5 03:20 /opt/compiler-explorer/cmake-v3.27.9
drwxr-xr-x 6 root root 6144 Dec 10 19:53 /opt/compiler-explorer/cmake-v3.28.0
drwxr-xr-x 6 root root 6144 Dec  5 03:23 /opt/compiler-explorer/cmake-v3.28.0-rc5
drwxr-xr-x 6 root root 6144 Apr 30 04:20 /opt/compiler-explorer/cmake-v3.28.4
drwxr-xr-x 6 root root 6144 Apr 30 04:24 /opt/compiler-explorer/cmake-v3.29.2
drwxr-xr-x 6 root root 6144 May 17 04:18 /opt/compiler-explorer/cmake-v3.29.20240506-g1ea8fa8
```

File: etc/config/cmakescript.amazon.properties (hunk 1 above)
Comment (mathstuf) [reply etc/config/cmakescript.amazon.properties:13]: Ah. The idea was to give the option of using "the latest CMake 3.x" option. I'll remove them here.

**Comment by mattgodbolt** [comment]
```
Found 1 property file(s) with issues:
################
## cmakescript.amazon.properties
bad_compilers_id:
  Line 1: cmake-3.29.20240506-g1ea8fa8
  Line 24: cmake-3
```

still some broken properties here.

**Comment by partouf** [comment]
now live

**Pull request merged**